test/models/ToolpadEditor.ts

```diff
@@ -137,23 +137,22 @@ export class ToolpadEditor {
 
     await this.componentCatalog.hover();
 
-    let pageRootBoundingBox;
-    await expect(async () => {
-      pageRootBoundingBox = await this.pageRoot.boundingBox();
-      expect(pageRootBoundingBox).toBeTruthy();
-    }).toPass();
+    await expect.poll(() => this.pageRoot.boundingBox()).toBeTruthy();
```

Comment:
I added a utility for this in [test/utils/waitForBoundingBox.ts](https://github.com/mui/mui-toolpad/blob/5a40939a4a8ee5fc5dbc43545d594ad1f1c172e5/test/utils/waitForBoundingBox.ts), maybe should separate it from the PR.
I can probably change it to use `.poll` too.

Reply:
I'm just allergic to the null-assertion operator 😄 
reverted this for now